addons/survey/controllers/survey_session_manage.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import werkzeug
+from werkzeug.urls import url_join
+
+from odoo import http
+from odoo.http import request
+
+
+class UserInputSession(http.Controller):
+    def _fetch_from_token(self, survey_token):
+        """ Check that given survey_token matches a survey 'access_token'.
+        Unlike the regular survey controller, user trying to access the survey must have full access rights! """
+        return request.env['survey.survey'].search([('access_token', '=', survey_token)])
+
+    @http.route('/survey/session_manage/<string:survey_token>', type='http', auth='user', website=True)
+    def survey_session_manage(self, survey_token, **kwargs):
+        """ Main route used by the host to 'manager' the session.
+        - If the state of the session is 'ready'
+          We render a template allowing the host to showcase the different options of the session
+          and to actually start the session.
+        - If the state of the session is 'in_progress'
+          We render a template allowing the host to show the question results, display the attendee
+          ranking or go to the next question of the session. """
+
+        survey = self._fetch_from_token(survey_token)
+
+        if not survey or survey.session_state == 'not_started':
+            # no open session
+            return werkzeug.utils.redirect('/')
```

Comment:
What do you think of a 404 ?

Reply:
I think it's a very good idea.
I think your feedback adds a great value to my work.
I think you're a wonderful colleague and I'm very glad to have the privilege of working with you.

(I think I need some "right-side-of-the-parking" time)